middleware/authorization.js

```diff
@@ -10,6 +10,9 @@ function verifyRole(roleName) {
       if (!userId) {
         return respondWithError(res, 401, "User not authenticated");
       }
+      // Check if user is admin first
+      const isAdmin = await repo.userIsAdmin(userId);
```

Comment:
maybe we could avoid doing two database calls to the roles table on this check.
can we fetch all user roles and check if admin or `roleName` is present in the result?

Reply:
@vm2052 I mean that if isAdmin check fails, we proceed to make another call to the database to get one more role.
wouldn't it be simpler if we did one call and checked all the roles we're interested in afterwards?

since this authorization check is performed quite often, we should think of making it faster and not overload it with unnecessary external calls